packages/jaeger-ui/src/reducers/deep-dependency-graph.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { handleActions } from 'redux-actions';
+
+import { actionTypes } from '../actions/deep-dependency-graph';
+import { fetchDeepDependencyGraph } from '../actions/jaeger-api';
+import { fetchedState } from '../constants';
+import { ApiError } from '../types/api-error';
+import transformDdgData from '../model/ddg/transformDdgData';
+import {
+  stateKey,
+  TDdgActionMeta,
+  TDdgAddStylePayload,
+  TDdgClearStylePayload,
+  TDdgPayload,
+  TDdgState,
+  TDdgStateEntry,
+} from '../model/ddg/types';
+import guardReducer, { guardReducerWithMeta } from '../utils/guardReducer';
+
+export function addStyleState(state: TDdgState, { payload }: { payload: TDdgAddStylePayload }) {
+  const { service, operation, start, end, visibilityIndices, style } = payload;
+  const key = stateKey(service, operation, start, end);
+  const stateEntry: TDdgStateEntry | undefined = state[key];
+  if (!stateEntry || stateEntry.state !== fetchedState.DONE) {
+    console.warn('Cannot set style state for unloaded Deep Dependency Graph'); // eslint-disable-line no-console
+    return state;
+  }
+  const styleStates = new Map(stateEntry.styleStates);
+  visibilityIndices.forEach(idx => {
+    styleStates.set(idx, (styleStates.get(idx) || 0) | style); // eslint-disable-line no-bitwise
```

Comment:
TBH I'm not totally consistent one way or another. I generally keep it on the same line unless the `eslint-disable` would cause the "target line" to exceed the length (i.e.:, instead of disabling `max-len`, i'd use `next-line`), or it is difficult to put the the `disable-line` on the same line (e.g.: needing to excuse a react prop it may be easiest to do:
```
{ /* eslint-disable-next-line blah */ }
<SomeComponent problemProp={problemValue} />
```
(but it may be even better to do:
```
<SomeComponent
  okayProp={okayValue}
  problemProp={problemValue /* eslint-disable-line blah */}
  alsoOkayProp={alsoOkayValue}
/>
```
))

Reply:
I was thinking 

```ts

  if (!stateEntry || stateEntry.state !== fetchedState.DONE) {
    // eslint-disable-next-line no-console
    console.warn('Cannot set view modifiers for unloaded Deep Dependency Graph'); 
    return state;
  }
```

Is more readable than

```ts
  if (!stateEntry || stateEntry.state !== fetchedState.DONE) {
    console.warn('Cannot set view modifiers for unloaded Deep Dependency Graph'); // eslint-disable-line no-console
    return state;
  }
```